src/lib/drivers/smbus_sbs/SBS.hpp

```diff
@@ -0,0 +1,325 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file SBS.hpp
+ *
+ * Header for a core SBS 1.1 specification of Smart Battery
+ * http://sbs-forum.org/specs/sbdat110.pdf
+ *
+ * @author Eohan George <eohan@rotoye.com>
+ * @author Nick Belanger <nbelanger@mail.skymul.com>
+ */
+
+#pragma once
+
+#include <px4_platform_common/i2c_spi_buses.h>
+#include <lib/drivers/smbus/SMBus.hpp>
+#include <uORB/topics/battery_status.h>
+#include <px4_platform_common/param.h>
+#include <ecl/geo/geo.h>
+
+
+using namespace time_literals;
+
+#define SBS_MEASUREMENT_INTERVAL_US             	100_ms          ///< time in microseconds, measure at 10Hz (not part of SBS spec)
+
+#define BATT_SMBUS_TEMP                                 0x08            ///< temperature register
+#define BATT_SMBUS_VOLTAGE                              0x09            ///< voltage register
+#define BATT_SMBUS_CURRENT                              0x0A            ///< current register
+#define BATT_SMBUS_AVERAGE_CURRENT                      0x0B            ///< average current register
+#define BATT_SMBUS_MAX_ERROR                            0x0C            ///< max error
+#define BATT_SMBUS_RELATIVE_SOC                         0x0D            ///< Relative State Of Charge
+#define BATT_SMBUS_ABSOLUTE_SOC                         0x0E            ///< Absolute State of charge
+#define BATT_SMBUS_REMAINING_CAPACITY                   0x0F            ///< predicted remaining battery capacity as a percentage
+#define BATT_SMBUS_FULL_CHARGE_CAPACITY                 0x10            ///< capacity when fully charged
+#define BATT_SMBUS_RUN_TIME_TO_EMPTY                    0x11            ///< predicted remaining battery capacity based on the present rate of discharge in min
+#define BATT_SMBUS_AVERAGE_TIME_TO_EMPTY                0x12            ///< predicted remaining battery capacity based on the present rate of discharge in min
+#define BATT_SMBUS_CYCLE_COUNT                          0x17            ///< number of cycles the battery has experienced
+#define BATT_SMBUS_DESIGN_CAPACITY                      0x18            ///< design capacity register
+#define BATT_SMBUS_DESIGN_VOLTAGE                       0x19            ///< design voltage register
+#define BATT_SMBUS_MANUFACTURER_NAME                    0x20            ///< manufacturer name
+#define BATT_SMBUS_MANUFACTURER_NAME_SIZE               21              ///< manufacturer name data size (not part of SBS spec)
+#define BATT_SMBUS_MANUFACTURE_DATE                     0x1B            ///< manufacture date register
+#define BATT_SMBUS_SERIAL_NUMBER                        0x1C            ///< serial number register
+#define BATT_SMBUS_MANUFACTURER_ACCESS                  0x00
+#define BATT_SMBUS_MANUFACTURER_DATA                    0x23
```

Comment:
I think it is better to define constants like this as either enum (for registers) or constexpr members since they obey scoping rules